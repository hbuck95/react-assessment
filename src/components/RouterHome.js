import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from "react-router-dom";

import NavigationBar from './NavigationBar';
import RegisterForm from './RegisterForm';
import LoginForm from './LoginForm';
import UserList from './UserList';

export default class RouterHome extends Component {
    constructor(props) {
        super(props);

        this.state = {
            activeUser: undefined
        };
    }

    render() {
        return (
            <div>
                <Router>
                    <NavigationBar/>
                    <Route exact path="/" render={() => <div><br/><p>Please register or login to your account.</p></div>} />
                    <Route path="/login" render={() => <LoginForm/>} />
                    <Route path="/register" render={() => <RegisterForm/>} />
                    <Route path="/viewUsers" render={() => <UserList/>} />
                </Router>
            </div>
        );
    }
}