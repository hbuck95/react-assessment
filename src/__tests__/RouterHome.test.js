import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';

import RouterHome from '../components/RouterHome';

configure({adapter: new Adapter()});

it('RouterHome renders', () => {
    const compJson = renderer.create(<RouterHome/>).toJSON();
    expect(compJson).toMatchSnapshot();
  });
