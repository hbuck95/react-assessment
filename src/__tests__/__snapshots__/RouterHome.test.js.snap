// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation bar renders 1`] = `
<div>
  <div
    id="Navbar"
  >
    <nav
      className="navbar navbar-expand-md navbar-light bg-dark"
    >
      <a
        className="navbar-brand"
      >
         
        <a
          className="nlink"
          href="/"
          onClick={[Function]}
        >
          React Assessment
        </a>
      </a>
      <button
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="ml-auto navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nlink"
              href="/register"
              onClick={[Function]}
            >
              Register
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nlink"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nlink"
              href="/viewUsers"
              onClick={[Function]}
            >
              View Users
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <div>
    <br />
    <p>
      Please register or login to your account.
    </p>
  </div>
</div>
`;

exports[`RouterHome renders 1`] = `
<div>
  <div
    id="Navbar"
  >
    <nav
      className="navbar navbar-expand-md navbar-light bg-dark"
    >
      <a
        className="navbar-brand"
      >
         
        <a
          className="nlink"
          href="/"
          onClick={[Function]}
        >
          React Assessment
        </a>
      </a>
      <button
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="ml-auto navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nlink"
              href="/register"
              onClick={[Function]}
            >
              Register
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nlink"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nlink"
              href="/viewUsers"
              onClick={[Function]}
            >
              View Users
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <div>
    <br />
    <p>
      Please register or login to your account.
    </p>
  </div>
</div>
`;
